### PR TITLE
Import net-misc/tinyssh from GURU

### PR DESCRIPTION
**Pull request by lanodan**

### CODE REVIEW

**Comment by lanodan** [comment]
btw `tinyssh` is server-only so it seems unfit to be added in `virtual/ssh`.

**Comment by lanodan** [comment]
>:=, as libsodium has a subslot.

Fixed

File: net-misc/tinyssh/tinyssh-99999999.ebuild [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+	default
+}
+
+src_compile() {
+	if use sodium
+	then
+		emake \
+			CC="$(tc-getCC)"
+			LIBS="-lsodium" \
```
Comment (thesamesam) [inline net-misc/tinyssh/tinyssh-99999999.ebuild:47]: Let's use pkg-config ($(tc-getPKG_CONFIG) ...) for LIBS, CFLAGS, and LDFLAGS please.

File: net-misc/tinyssh/tinyssh-99999999.ebuild [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+			CFLAGS="${CFLAGS} -I/usr/include/sodium" \
```
Comment (thesamesam) [inline net-misc/tinyssh/tinyssh-99999999.ebuild:48]: We should use pkg-config, but also, _if_ we were keeping this, it should be `CPPFLAGS` and prefixed with `ESYSROOT`. But we're not :)

File: net-misc/tinyssh/tinyssh-99999999.ebuild [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+			LDFLAGS="${LDFLAGS} -L/usr/lib"
```
Comment (thesamesam) [inline net-misc/tinyssh/tinyssh-99999999.ebuild:49]: This isn't right - `/usr/lib` is not necessarily the native libdir (it's 32-bit on amd64), not right on prefix either, ut it's also unnecessary as it'll be in the linker search path anyway. Just drop that -L ... bit.

File: net-misc/tinyssh/tinyssh-20230101.ebuild [hunk 4]
```diff
@@ -0,0 +1,71 @@
+# Copyright 1999-2023 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
```
[the remark this answers is not in the document]
Comment (thesamesam) [reply net-misc/tinyssh/tinyssh-20230101.ebuild:4]: ^

**Comment by thesamesam** [comment]
Oh, I didn't mean to push this. Can you handle the comments in a followup rather than me revert it though?

**Comment by lanodan** [comment]
>Oh, I didn't mean to push this. Can you handle the comments in a followup rather than me revert it though?

Done at https://github.com/gentoo/gentoo/pull/32045

**Pull request closed without merging**